src/opencensus.erl

```diff
@@ -54,16 +63,55 @@
 -type network_event() :: #network_event{}.
 -type network_event_type() :: recv | sent | unspecified.
 
--type maybe(T)        :: T | undefined.
+-type maybe(T) :: T | undefined.
 
 %% timestamp in microseconds
 -type time_us() :: non_neg_integer().
 
 %%--------------------------------------------------------------------
 %% @doc
+%% Creates a new trace context if `enabled` is true in the trace context
+%% argument or the sampling function returns true. If the sampling returns
+%% false then `undefined` is returned.
+%% @end
+%%--------------------------------------------------------------------
+-spec start_trace() -> trace_context().
+start_trace()  ->
+    start_trace(generate_trace_id(), undefined, undefined).
+
+start_trace(undefined)  ->
+    start_trace(generate_trace_id(), undefined, undefined);
+start_trace(#trace_context{trace_id=TraceId,
+                           span_id=SpanId,
+                           enabled=Enabled})  ->
+    start_trace(TraceId, SpanId, Enabled);
+start_trace(TraceId)  ->
+    start_trace(TraceId, undefined, undefined).
+
+start_trace(TraceId, SpanId, true)  ->
+    #trace_context{trace_id = TraceId,
+                   span_id = SpanId,
+                   enabled = true};
+start_trace(TraceId, SpanId, _)  ->
```

Comment:
even if this is the catch all for the sampling function, shouldn't there be a clause for false? or will false mean handled by the sampling function?

based on the [specs for the header](https://github.com/TraceContext/tracecontext-spec/blob/master/HTTP_HEADER_FORMAT.md#trace-options) the enabled flag is just a suggestion, soperhaps we should always set it to 'what the caller passed' either true or false and then let the sampler decide weather they want to honor it or not?

Reply:
If it is `false` then the sampling function is used, while true means it is suggested that it should be sampled:

>Controls tracing options such as sampling, trace level etc. It is a 1-byte representing a 8-bit unsigned integer. The least significant bit provides recommendation whether the request should be traced or not (1 recommends the request should be traced, 0 means the caller does not make a decision to trace and the decision might be deferred). The flags are recommendations given by the caller rather than strict rules to follow for 3 reasons:
>
>    Trust and abuse.
>    Bug in caller
>    Different load between caller service and callee service might force callee to down sample.

Reply:
yes suggests, the last sentence is pretty clear:

The flags are **recommendations** given by the caller rather than strict rules ...

Given the specs, neither true nor false should be enforced but rather passed into the sampler and let the sampler decide whether to follow the recommendation or not.

Reply:
Oh, I think I get what you suggested for the change. I was confused because in my mind I knew that the `true` being matched on and setting it to `true` would have to change but that it didn't matter right now since there are no samplers :). 

But yea, I can change it to just be one function clause of `Enabled` until the samplers are created.